src/viper/trans.ml

```diff
@@ -643,6 +671,19 @@ and assign_stmts ctxt at (lval : lvalue) (e : M.exp) : seqn' =
   match e with
   | M.({it=TupE [];_}) -> [], []
   | M.({it=AnnotE (e, _);_}) -> assign_stmts ctxt at lval e
+  | M.({it=CallE ({it=M.DotE ({it=M.VarE(m);_}, {it="init";_});_}, _inst, args);_})
+      when Imports.find_opt (m.it) ctxt.imports = Some(IM_base_Array)
+      ->
+    begin match args with
+    | M.({it=TupE([e1;e2]); _}) ->
+      fld_via_tmp_var lval t (fun x ->
+        let lhs = !!(LocalVar (x, tr_typ t)) in
+        [], [ !!(InhaleS (array_acc at lhs (array_elem_t t)))
+            ; !!(InhaleS (array_size_inv at lhs (exp ctxt e1)))
+            ; !!(InhaleS (array_init_const at lhs (array_elem_t t) (exp ctxt e2))) ]
```

Comment:
```suggestion
        [], [ !!(ExhaleS !!(LeCmpE (intLitE at 0) (exp ctxt e1) )
            ; !!(InhaleS (array_acc at lhs (array_elem_t t)))
            ; !!(InhaleS (array_size_inv at lhs (exp ctxt e1)))
            ; !!(InhaleS (array_init_const at lhs (array_elem_t t) (exp ctxt e2))) ]
```
I have added check that we initializing array by positive length (didn't compile)

Reply:
After it I think you can merge it